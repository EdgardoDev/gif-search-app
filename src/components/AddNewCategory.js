import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddNewCategory = ({ setAppCategories }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    // Small validation before submitting.
    if (inputValue.trim().length > 2) {
      setAppCategories(categories => [ inputValue, ...categories ]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={ handleSubmit }>
      <input
        type="text"
        value={inputValue}
        onChange={ handleInputChange }
      />
    </form>
  )
}

// PropTypes definition.
AddNewCategory.propTypes = {
  setAppCategories: PropTypes.func.isRequired,
}

