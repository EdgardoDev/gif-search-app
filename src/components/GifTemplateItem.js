import React from 'react'
import PropTypes from 'prop-types';

export const GifTemplateItem = ({ title, url }) => {
  return (
    <div className="gifs-card animate__animated animate__bounce">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

// PopTypes implementation.
GifTemplateItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}


