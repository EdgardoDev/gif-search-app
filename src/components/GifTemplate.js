import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifTemplateItem } from './GifTemplateItem';

export const GifTemplate = ({ appCategory }) => {

  const { data:images, loading } = useFetchGifs(appCategory);

  return (
    <>
      <h3 className="category">{appCategory}</h3>

      { loading && <p>Loading...</p> }
      
      <div className="gifs-container">
        {
          images.map(img => (
            <GifTemplateItem
              key={img.id}
              {...img}
            />
          ))
        }
      </div>
    </>
  )
}
