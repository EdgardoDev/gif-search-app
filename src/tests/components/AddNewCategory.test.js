import React from 'react';
import { shallow } from 'enzyme';
import { AddNewCategory } from '../../components/AddNewCategory';

describe('Tests for the component <AddNewCategory />', () => {

  const setAppCategories = () => { };
  const wrapper = shallow(<AddNewCategory setAppCategories={setAppCategories} />)

  test('It should render without any problems.', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('It should evaluate changes on the input', () => {
    const input = wrapper.find('input');
    const value = 'Hello World!';
    input.simulate('change', { target: { value } });

    expect(wrapper.find('p').text().trim()).toBe(value);
  })

})