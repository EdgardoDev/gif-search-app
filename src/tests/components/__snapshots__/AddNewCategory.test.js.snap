// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the component <AddNewCategory /> It should render without any problems. 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
